File: api-reseach/rdw.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import os
os.system('clear')
import sys
from bs4 import BeautifulSoup
import requests


if len(sys.argv) == 1:
  kenteken = "21-snz-3"
elif sys.argv[1] == 'volvo':
  kenteken = "21-SNZ-3"
elif sys.argv[1] == 'renault':
  kenteken = ""
else :
  kenteken = sys.argv[1]

headers = {
    'authority': 'ovi.rdw.nl',
    'accept': 'text/html, */*; q=0.01',
    'x-requested-with': 'XMLHttpRequest',
    'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.89 Safari/537.36',
    'content-type': 'application/x-www-form-urlencoded; charset=UTF-8',
    'origin': 'https://ovi.rdw.nl',
    'sec-fetch-site': 'same-origin',
    'sec-fetch-mode': 'cors',
    'sec-fetch-dest': 'empty',
    'referer': 'https://ovi.rdw.nl/',
    'accept-language': 'nl-NL,nl;q=0.9,en-US;q=0.8,en;q=0.7',
    '$cookie': 'ASP.NET_SessionId=51mzz4dioadyxqtgmexbckep; rdw-persist=\\u0021jkroQeZBF2fKDl6W5/spYcFGPNu8GnDZ5MWVW1YkRi/aG4T/eFWhTkLZm0ZlX8RLArbIBAAc2UlF6tpaIb7k2n9xXE736y9XvJgJYXu+qQ==; f5avraaaaaaaaaaaaaaaa_session_=DCPBPNGMHMMGFDAFGEHJKFOJMBHKOEDNAFAKJDMLLIAOIBLIBPMFDKKEBOABIMNLGNIDEABLOJEOFKCPJHJAHGJDKKPDHLCDGHLPKHLGALEIMPPIAPLDJGILFINNEKBM; TS011f44e9=01a5d145de301b7405b2208d0d4254e0744ff471408bee22b7c8351eaa7ad087d70a84b7e456281cfdceb8644f93287058d26a3f9f10f312a1ec998dae6309b19a930d536ca2bd733ab799426a6873aae5a308129e2056e8e4a78142b6177719397cfbefafff5921121e7e81f5e2c8dafabbcdc9cc403328144c5aac8fd0940725eaaccf2b; f5avr1881096869aaaaaaaaaaaaaaaa_cspm_=PDPIDGCMPECPELKKCLLHKLJFCHODAFOHICFIPOEFLBDDHLNIJHMJMOGEPLCNAIBGKNHCDENMDPLEKLPFCNBAJHBLALFLOACJLPHLNOLAALEHGCCDOFKFFNEFEMBEFDIB',
}

data = {
  '__VIEWSTATE': '/wEPDwUKMTE1NDI3MDEyOQ9kFgJmD2QWAgIDD2QWAgIBD2QWAgIJDxYCHgdWaXNpYmxlaGRkbswJ9j6e1dGi3igmthIyvj8gc4WH7JTnjgF14yLn50Y=',
  '__VIEWSTATEGENERATOR': 'CA0B0334',
  '__EVENTVALIDATION': '/wEdAAK8JKqL+RK4NWrMB2xu8rTB851Fq81QBiZgFEttEk2eeHewUOoCJ2ceZf9ZQoBBMXI2OTBKyEO6GXbPjhMnQTm0',
  'ctl00$TopContent$txtKenteken': kenteken
}

page = requests.post('https://ovi.rdw.nl/', headers=headers, data=data)
soup = BeautifulSoup(page.content, 'html.parser')


merk = soup.find("div", class_="ui-block-a")

print("het is een ", str(merk.find("h3").get_text()))

print("En het model is ", str(merk.find("p").get_text()))






import os
os.system('clear')
import sys
from bs4 import BeautifulSoup
import requests


if len(sys.argv) == 1:
  kenteken = "21-snz-3"
elif sys.argv[1] == 'volvo':
  kenteken = "21-SNZ-3"
elif sys.argv[1] == 'renault':
  kenteken = ""
else :
  kenteken = sys.argv[1]


data = {
  '__VIEWSTATE': '/wEPDwUKMTE1NDI3MDEyOQ9kFgJmD2QWAgIDD2QWAgIBD2QWAgIJDxYCHgdWaXNpYmxlaGRkbswJ9j6e1dGi3igmthIyvj8gc4WH7JTnjgF14yLn50Y=',
  '__VIEWSTATEGENERATOR': 'CA0B0334',
  '__EVENTVALIDATION': '/wEdAAK8JKqL+RK4NWrMB2xu8rTB851Fq81QBiZgFEttEk2eeHewUOoCJ2ceZf9ZQoBBMXI2OTBKyEO6GXbPjhMnQTm0',
  'ctl00$TopContent$txtKenteken': kenteken
}

page = requests.post('https://ovi.rdw.nl/', data=data)
soup = BeautifulSoup(page.content, 'html.parser')


merk = soup.find("div", class_="ui-block-a")

print("het is een ", str(merk.find("h3").get_text()))

print("En het model is ", str(merk.find("p").get_text()))

File: api-reseach/rdw.js
let JSSoup = require('jssoup').default;
const fetch = require('node-fetch');
fetch("https://ovi.rdw.nl/", {
    "headers": {
      "accept": "text/html, */*; q=0.01",
      "accept-language": "nl-NL,nl;q=0.9,en-US;q=0.8,en;q=0.7",
      "content-type": "application/x-www-form-urlencoded; charset=UTF-8",
      "sec-fetch-dest": "empty",
      "sec-fetch-mode": "cors",
      "sec-fetch-site": "same-origin",
      "x-requested-with": "XMLHttpRequest",
      "cookie": "ASP.NET_SessionId=5ydd4o4zoluvvxkqkbrttsnv; rdw-persist=!lUEn6uvqFJgcszyW5/spYcFGPNu8GhS1pCDT7t3s+QWZyWuuiGFDeyytv+UgiJkNa7JuKIJjxV9q490SfscBGWdW0IChdrAH9LzEQJunjw==; f5avraaaaaaaaaaaaaaaa_session_=MNMPNBIBACEOOLACIOLPKHJKCBHGGELFLJBNMFJDMMPLHENOHIHBOGCEBFAGCOMAHAKDNMINFMBDLFENDAMANHFPMFEKCEGMEPPBEJEAFDMMHJHMKLKJCNNEDOKFMHMG; TS011f44e9=01a5d145de5b790e4442a85d6c4ef3c3b8bcc463b1a85cb2a079fa75bd7bede9686e368c14540b2dd926775b73560a425e1924bc5dc7c873b861c9e984b03ef4ef947a2b3e737fe36b607f2c0fd2b793a19ce9e6fa345620d096bd7335fb9404e2b01aa001; _ga=GA1.3.1567136690.1596611859; _gid=GA1.3.464048628.1596611859; _gat_UA-123561920-5=1"
    },
    "body": `__VIEWSTATE=%2FwEPDwUKMTE1NDI3MDEyOQ9kFgJmD2QWAgIDD2QWAgIBD2QWAgIJDxYCHgdWaXNpYmxlaGRkbswJ9j6e1dGi3igmthIyvj8gc4WH7JTnjgF14yLn50Y%3D&__VIEWSTATEGENERATOR=CA0B0334&__EVENTVALIDATION=%2FwEdAAK8JKqL%2BRK4NWrMB2xu8rTB851Fq81QBiZgFEttEk2eeHewUOoCJ2ceZf9ZQoBBMXI2OTBKyEO6GXbPjhMnQTm0&ctl00%24TopContent%24txtKenteken=${process.env["KENTEKEN"]}`,
    "method": "POST",
    "mode": "cors"
  }).then(res => res.text().then(res => hadany(res)))
  .then(res => console.log(res))
.catch(err => console.log("hij ging fout", err));

function hadan(res) {
  var soup = new JSSoup(res);
  let merk = soup.find("div", class_="ui-block-a")
  console.log("het is een ", merk.find("h3").text);
  console.log("En het model is ", merk.find("p").text);
    }
  function hadany(res) {
    var soup = new JSSoup(res);
    if (soup.find("div", class_="text-red")) {
      let fout_melding = soup.find("div", class_="text-red").text
      return fout_melding;
    }
    else {
      let merk = soup.find("div", class_="ui-block-a")
      console.log("het is een ", merk.find("h3").text);
      console.log("En het model is ", merk.find("p").text);
      return "het is een ", merk.find("h3").text, "En het model is ", merk.find("p").text;
    }
  }
